packages/visual-editor/src/hooks/useReadOnlyComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
```

Comment:
[6] Mostly copy-pasted code from the `useEditorSubscriber`, excluding all the events that we are not interested in, like the drag and drop stuff, hover or select events. 

this will mostly just listen to the component tree update events and will update the tree in here